besthouse/client/main.go

```
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+
+	"github.com/mlgx/puzzling/besthouse"
+)
+
+func mapGenerate(mReq besthouse.MapReq) (*besthouse.MapRes, error) {
+	buf, err := json.Marshal(mReq)
+	if err != nil {
+		return nil, err
+	}
+	body := bytes.NewReader(buf)
+	req, err := http.NewRequest("GET", "http://127.0.0.1:8080/generate", body)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode != http.StatusOK {
+		s, _ := ioutil.ReadAll((resp.Body))
+		return nil, fmt.Errorf("Server Says: %d %s", resp.StatusCode, s)
+	}
+	mRes := &besthouse.MapRes{}
+	if err := json.NewDecoder(resp.Body).Decode(mRes); err != nil {
+		if err != io.EOF {
+			return nil, err
+		}
+	}
+	return mRes, nil
+}
+
+func mapValidate(mVal besthouse.MapValidateReq) (bool, error) {
+	buf, err := json.Marshal(mVal)
+	if err != nil {
+		return false, err
+	}
+	body := bytes.NewReader(buf)
+	req, err := http.NewRequest("GET", "http://127.0.0.1:8080/validate", body)
+	if err != nil {
+		return false, err
+	}
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return false, err
+	}
+	if resp.StatusCode != http.StatusOK {
+		s, _ := ioutil.ReadAll((resp.Body))
```

Need comment on ignored error.